web/src/engine/websites/NextScan.ts

```diff
@@ -0,0 +1,77 @@
+import { Tags } from '../Tags';
+import icon from './NextScan.webp';
+import { Chapter, DecoratableMangaScraper, type MangaPlugin, Manga } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON, FetchRequest, FetchWindowScript } from '../FetchProvider';
+
+const chapterScript = `
+    new Promise(resolve => {
+        const mangatitle = document.querySelector('article header h1').textContent.trim();
+        const chapters = epX.arr
+            .filter(chapter => chapter.url != window.location.href)
+            .map(chapter => {
+                return {
+                    id : new URL(chapter.url).pathname,
+                    title : chapter.title.replace(mangatitle, '').trim()
+                }
+            });
+
+        resolve(chapters);
+    });
+`;
+
+const pagesScript = `
+    new Promise(resolve => {
+        resolve([...document.querySelectorAll('div#readerarea img')].map(img => img.src));
+    });
+`;
+
+type ChapterID = {
+    id: string,
+    title: string
+}
+
+type APIMangas = {
+    feed: {
+        entry: {
+            title: {
+                $t : string
+            },
+            link: {
+                rel: string
+                title: string,
+                href: string
+            }[]
+        }[]
+    }
+}
+
+@Common.MangaCSS(/^{origin}\/\d+\/\d+\/[^/]+\.html$/, 'article header h1')
+@Common.PagesSinglePageJS(pagesScript, 500)
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
```

Comment:
Can't `Common.ChaptersSinglePageJS` decorator be used?

Reply:
True, what was i thinking?